tests/Progress/SnapshotTest.php

```diff
@@ -38,6 +39,42 @@ public function ofActivityInstance_builds_a_module_instance_progress_for_each_mo
         $this->assertContainsOnlyInstancesOf(ModuleInstanceProgress::class, $progress->getModules());
     }
 
+    /** @test */
+    public function ofUpdatesToActivityInstance_only_returns_progress_when_updated()
```

Comment:
Could there be a unit version of this, mock the repository to make sure it's using the API correctly and tells us if problems are in the Snapshot class or repository